### PR TITLE
Joystick: Allow for background events; Add deadzone to SDLAnalog

### PR DESCRIPTION
**Pull request by B3n30**
This adds two features to the joystick code:

1. Events from joysticks will now get registered even if citra isn't in focus
2. Analog axis can have a deadzone. There is no gui for setting it yet. Adding a deadzone can be done via the config inis, e.g.:
`circle_pad="port:0,guid:050000005769696d6f74652028303000,axis_x:0,axis_y:1,engine:sdl,deadzone:0.5"`

the deadzone needs to be: `0.0 <= deadzone < 1.0`

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4193)

### CODE REVIEW

**Comment by wwylele** [comment]
What's the status of this?

**Comment by wwylele** [comment]
Please resolve conflict

**Comment by B3n30** [comment]
Any more reviews? The crash on appveyor is unrelated to that Pr

File: src/input_common/sdl/sdl_impl.cpp [hunk 1]
```diff
@@ -324,17 +324,26 @@ class SDLAxisButton final : public Input::ButtonDevice {
...
-    SDLAnalog(std::shared_ptr<SDLJoystick> joystick_, int axis_x_, int axis_y_)
-        : joystick(std::move(joystick_)), axis_x(axis_x_), axis_y(axis_y_) {}
+    SDLAnalog(std::shared_ptr<SDLJoystick> joystick_, int axis_x_, int axis_y_, float deadzone_)
+        : joystick(std::move(joystick_)), axis_x(axis_x_), axis_y(axis_y_), deadzone(deadzone_) {}
 
     std::tuple<float, float> GetStatus() const override {
-        return joystick->GetAnalog(axis_x, axis_y);
+        float x;
+        float y;
+        std::tie(x, y) = joystick->GetAnalog(axis_x, axis_y);
```
Comment (jroweboy) [inline src/input_common/sdl/sdl_impl.cpp]:
```cpp
float x;
float y;
std::tie(x, y) = joystick->GetAnalog(axis_x, axis_y);
```
can't this just be

```cpp
const auto[x, y] = joystick->GetAnalog(axis_x, axis_y);
```

now that we be cpp 17

**Pull request merged**